autoconfigure-adapter/src/main/java/org/springframework/boot/autoconfigure/jdbc/DataSourceType.java

```diff
@@ -0,0 +1,5 @@
+package org.springframework.boot.autoconfigure.jdbc;
+
+public enum DataSourceType {
```

Comment:
Even if that mean some kind of duplication, I think it would be better to not put user facing API used in the `autoconfigure-adapter` module since it is going to be removed at some point. This probably should be part of KoFu and JaFu DSL for now.